### PR TITLE
Add robust ignore

### PR DESCRIPTION
**Pull request by gavanderhoorn**
Should do what we discussed in #358.

@ChrisTimperley and @wasowski if you could take a look?

### CODE REVIEW

File: scripts/build-bugzoo.py [hunk 1]
```diff
@@ -11,10 +11,16 @@
...
 
+IGNORE_FILENAME='ROBUST_IGNORE'
 
-def find_bug_descriptions(d):
+def find_bug_descriptions(d, ignore_marker=IGNORE_FILENAME):
     buff = []
-    for root, _, files in os.walk(d):
+    for root, dirnames, files in os.walk(d):
+        if ignore_marker in files:
+            logger.info ("Ignoring {} (and children) because of ROBUST_IGNORE".format(root))
```
Comment (ChrisTimperley) [inline scripts/build-bugzoo.py:20]: space between `.info` and `(`.

File: scripts/build-rosinstall.py [hunk 2]
```diff
@@ -19,10 +19,16 @@
...
+IGNORE_FILENAME='ROBUST_IGNORE'
 
-def find_bug_descriptions(d):
+def find_bug_descriptions(d, ignore_marker=IGNORE_FILENAME):
     buff = []
-    for root, _, files in os.walk(d):
+    for root, dirnames, files in os.walk(d):
+        if ignore_marker in files:
+            logger.info ("Ignoring {} (and children) because of ROBUST_IGNORE".format(root))
+            dirnames[:] = []
```
Comment (ChrisTimperley) [inline scripts/build-rosinstall.py:29]: Same comment about deadcode as above.

File: scripts/build-bugzoo.py [hunk 3]
```diff
@@ -11,10 +11,16 @@
...
+            dirnames[:] = []
```
[the remark this answers is not in the document]
Comment (ChrisTimperley) [reply scripts/build-bugzoo.py:21]: Ah, I see. `dirnames` is used internally by `os.walk` to continue the walk.

File: scripts/build-rosinstall.py [hunk 4]
```diff
@@ -19,10 +19,16 @@
...
 
+IGNORE_FILENAME='ROBUST_IGNORE'
 
-def find_bug_descriptions(d):
+def find_bug_descriptions(d, ignore_marker=IGNORE_FILENAME):
     buff = []
-    for root, _, files in os.walk(d):
+    for root, dirnames, files in os.walk(d):
+        if ignore_marker in files:
+            logger.info ("Ignoring {} (and children) because of ROBUST_IGNORE".format(root))
```
[the remark this answers is not in the document]
Comment (gavanderhoorn) [reply scripts/build-rosinstall.py]: *Excess* space even ;)

But I've removed them.

File: scripts/build-bugzoo.py [hunk 5]
```diff
@@ -11,10 +11,16 @@
...
 
+IGNORE_FILENAME='ROBUST_IGNORE'
 
-def find_bug_descriptions(d):
+def find_bug_descriptions(d, ignore_marker=IGNORE_FILENAME):
     buff = []
-    for root, _, files in os.walk(d):
+    for root, dirnames, files in os.walk(d):
+        if ignore_marker in files:
+            logger.info("Ignoring {} (and children) because of ROBUST_IGNORE".format(root))
```
Comment (ChrisTimperley) [inline scripts/build-bugzoo.py:20]: If you want to be really clever, you could swap `ROBUST_IGNORE` with a template argument (i.e., `{}` and `IGNORE_FILENAME`).

Comment (gavanderhoorn) [reply scripts/build-bugzoo.py]: Initially `ignore_marker` was hard-coded to always be `ROBUST_IGNORE`, then I made it a defaulted arg but forgot to update the log message.

Should be fixed now.

**Review by ChrisTimperley — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**